rfc/0005-policy-reports.md

```diff
@@ -0,0 +1,110 @@
+| RFC PR       | [fill this in after opening PR] |
+
+# Summary
+[summary]: #summary
+
+> Brief (one-paragraph) explanation of the feature.
+
+Policy Reports are a lightweight form of policy result reporting that
+allows users to understand the impact of a policy on the cluster
+without the need to deploy the whole telemetry and observability
+stack.
+
+# Motivation
+[motivation]: #motivation
+
+> - Why are we doing this?
+>
+> Describe the problem you are trying to solve, and its constraints,
+> without coupling them too closely to the solution you have in
+> mind. If this RFC is not accepted, the motivation can be used to
+> develop alternative solutions.
+
+In order to understand how a policy behaves in a live cluster, it's
+possible to relay on traces, logs and metrics of the
+`policy-server`. However, if a user does not want to deploy the whole
+stack, but still have a good understanding on how a policy would
+behave or behaves in a live context, they can relay on this new
+feature -- Policy Reports.
+
+## Examples / User Stories
+[examples]: #examples
+
+> Examples of how the feature will be used. Interactions should show
+> the action and the response. When appropriate, provide user stories
+> in the form of "As a [role], I want [feature], so [that]."
+
+> As a user testing Kubewarden, I can check in a simple and
+> understandable way how a policy or set of policies behave -- what
+> is/are accepting, rejecting or mutating -- just by deploying
+> Kubewarden, without the need of deploying an observability stack.
+
+> As a user that does not want to deploy the whole stack, I can deploy
+> policies in `monitor` mode --so they don't impact my cluster
+> behavior--, and at the same time, check what decisions they would
+> have taken had they been in `protect` mode. Only when I'm sure they
+> won't wreak havoc in my cluster, I can promote them from `monitor`
+> to `protect`.
+
+> As a user, I am able to ask for a report of a specific policy or set
+> of policies and understand how they are behaving (regardless of they
+> deployment mode -- `monitor` or `protect`).
+
+> As a user, I can check if a settings change on a policy is impacting
+> the number of rejections or mutations the policy is resolving to in
+> a noticeable way.
+
+> As a user, I can understand if a policy is targeting a wider number
+> of requests than it should by looking at the evaluation number
+> without the need of deploying the observability stack.
+
+> As a UI integrator, I can show rich stats about evaluations without
+> the need of doing any computations, just by consuming reports, which
+> have a well known, stable and versioned structure.
+
+> As a third party integrator, the interface to fetch Kubewarden stats
+> is the well known and versioned Report resource. I can use this
+> information from Kubewarden to aggregate to other security relevant
+> data about the cluster in a hollistic security-focused control
+> plane.
```

Comment:
Does the user here expect low latency or pull behaviour, for automated/machine readable usecases? Maybe it is worth it to clarify.
For pull behaviour, we could always support setting annotation on the Report, that would trigger the controller to recompute it.

Reply:
The idea is that the report is updated by a component when requests arrive, or by buffering events and dumping them onto the report from time to time (e.g. every 30 seconds or 1 minute).

This resource is for informational purposes and for the consumer to have a general overview of what is happening/would happen with policies on the cluster.